src/validate.js

```diff
@@ -0,0 +1,36 @@
+import { isObject, toErrorList, mergeObjects } from "./utils";
+
+
+function createErrorHandler(formData) {
+  const handler = {
+    __errors: [],
+    addError(message) {
+      this.__errors = [...this.__errors, message];
+    },
+  };
+  if (isObject(formData)) {
+    return Object.keys(formData).reduce((acc, key) => {
+      return {...acc, [key]: createErrorHandler(formData[key])};
+    }, handler);
+  }
+  return handler;
+}
+
+function unwrapErrorHandler(errorHandler) {
+  return Object.keys(errorHandler).reduce((acc, key) => {
+    if (key === "addError") {
+      return acc;
+    } else if (key === "__errors") {
+      return {...acc, [key]: errorHandler[key]};
```

Comment:
Why? If the key is named `__errors` we want the accumulator to have it as well. Maybe that's just poor naming that confused you here? Or am I missing something else?